projects/extension/src/background/mocks.ts

```diff
@@ -40,21 +40,27 @@ export class MockPort implements chrome.runtime.Port {
 
 export class MockConnectionManager implements ConnectionManagerInterface {
   readonly #willFindClient: boolean;
+  lastId = 0;
+
 
   constructor(willFindClient: boolean) {
     this.#willFindClient = willFindClient;
   }
 
-  registerAppWithSmoldot(app: AppMediator, name: string) {
+  registerAppWithSmoldot(app: AppMediator, name: string): void {
```

Comment:
what `registerAppWithSmoldot`and `unregisterApp`do?

Reply:
They are book-keeping methods.. `registerAppWithSmoldot` tells the specified `SmoldotMediator` about a new app (it adds the app to its internal apps list) ... when incoming messages arrive from that smoldot we only ask the apps that have been registered with that smoldot to process that message (not all the apps as the others arent interested).  `unregisterApp` does the opposite, it tells the smoldot that that app is now disconnected so that we stop asking it to process messages that come from that smoldot.

Now I reread this I think that mnames should be symetrical :thinking:  `registerApp` and `unregisterApp`

Reply:
Ive renamed it for consistency